crates/core/src/contracts/contract_account.rs

```diff
@@ -58,4 +58,23 @@ impl<'a, P: Provider + Send + Sync> ContractAccount<'a, P> {
         let value = Into::<U256>::into(low) + (Into::<U256>::into(high) << 128);
         Ok(value)
     }
+
+    /// Returns the nonce of the contract account.
```

Comment:
I think since this method is a bit special, it deserves more context in the comment, 
- something explaining that there are two types on accounts in EVM, EOA and Contract Account. That EOA nonce is handled by Starknet protocol and that Contract Account nonce is handled by a dedicated storage 
- this function fetches the nonce in storage

Reply:
Gotcha adding this in the docstring